src/kerneltypes.h

```diff
@@ -31,4 +31,10 @@ static inline uint16_t hton16( uint16_t x )
 	return (x&0xff)<<8 | (x>>8);
 }
 
+struct process_info {
```

Comment:
Unless you have some really specific reason for needing integers of a certain size, I suggest just using ints.
(I think I got a little carried away with this earlier in the code.)